griptape/drivers/prompt/huggingface_hub_prompt_driver.py

```diff
@@ -61,11 +61,18 @@ def try_run(self, prompt_stack: PromptStack) -> TextArtifact:
         return TextArtifact(value=response)
 
     def try_stream(self, prompt_stack: PromptStack) -> Iterator[TextArtifact]:
-        prompt = self.prompt_stack_to_string(prompt_stack)
+        prompt = self.__to_prompt(prompt_stack)
 
         response = self.client.text_generation(
             prompt, return_full_text=False, max_new_tokens=self.max_output_tokens(prompt), stream=True, **self.params
         )
 
         for token in response:
             yield TextArtifact(value=token)
+
+    def __to_prompt(self, prompt_stack: PromptStack) -> str:
+        tokens = self.tokenizer.tokenizer.apply_chat_template(
+            [{"role": i.role, "content": i.content} for i in prompt_stack.inputs], add_generation_prompt=True
+        )
+
+        return self.tokenizer.tokenizer.decode(tokens)
```

Comment:
`apply_chat_template` returns a string directly if you pass `tokenize=False`

Reply:
Setting up for a future PR that will use the tokens.